File: src/LeetCode/Algorithms/Subtract_the_Product_and_Sum_of_Digits_of_an_Integer.java
package LeetCode.Algorithms;

// https://leetcode.com/problems/subtract-the-product-and-sum-of-digits-of-an-integer/
public class Subtract_the_Product_and_Sum_of_Digits_of_an_Integer {
	public static void main(String[] args) {
		int n = 234;

		System.out.println(subtractProductAndSum(n));
	}

	public static int subtractProductAndSum(int n) {
		int multi = 1;
		int sum = 0;
		int rest = 0;

		while(n > 0) {
			rest = n % 10;
			n /= 10;

			multi *= rest;
			sum += rest;
		}

		return multi - sum;
	}
}

File: src/LeetCode/Algorithms/JavaScript/Subtract_the_Product_and_Sum_of_Digits_of_an_Integer.js
/**
 * https://leetcode.com/problems/subtract-the-product-and-sum-of-digits-of-an-integer/
 * @param {number} n
 * @return {number}
 */
var subtractProductAndSum = function(n) {
	let product = 1;
	let sum = 0;
	let rest = 0;
	
	while (n > 0) {
		rest = n % 10;
		n = parseInt(n / 10);
		
		product *= rest;
		sum += rest;
	}
	
	return (product - sum);
};

let n = 4421;

console.log(subtractProductAndSum(n));